### PR TITLE
archive.extracted taints --retcode-passthrough when used for tar and called with options

### PR DESCRIPTION
**Pull request by ahus1**
The `archive.extrated` will return the `cmd.retcode` return code of `1` on for the a `salt-call state.highstate --retcode-passthrough` when called in a Gnu tar environment with tar options, although the extraction worked fine.

I our case this lead to a scenario with Vagrant's Salstack provisioner to assume that provisioning has failed.

I provide the change and a unit test. Please let me know if the PR is missing anything.

### CODE REVIEW

File: salt/states/archive.py [hunk 1]
```diff
@@ -211,7 +212,7 @@ def extracted(name,
                 ret['result'] = False
                 ret['changes'] = results
                 return ret
-            if __salt__['cmd.retcode']('tar --version | grep bsdtar', python_shell=True) == 0:
+            if re.search('bsdtar', __salt__['cmd.run']('tar --version', python_shell=True)):
```
Comment (jfindlay) [inline salt/states/archive.py]: I would do
```python
if 'bsdtar' in  __salt__['cmd.run']('tar --version'):
```
Other than that, this looks great.  Also, @thatch45 may condemn me for picking at code thus.  :-)

**Comment by jfindlay** [comment]
@ahus1, there are some [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/1843/violations/file/tests/unit/states/archive_test.py/), but thanks for providing tests, that is awesome.

**Comment by ahus1** [comment]
@jfindlay I changed the code and eliminated the lint errors. 

I plead *not guilty* regarding the failed cent7 run ("Connection reset by peer"), the ubuntu test ("SSHCustomModuleTest" is not mine), and pr-lint-n ("boto_route53.py" is not a module I touched). 

Do you consider this ready for merge? Thanks.

**Comment by jfindlay** [comment]
@ahus1, I agree.  The CentOS 7 tests didn't even run, but your new tests did run on the ubuntu 14.04 run.

**Pull request merged**